tutor/src/screens/student-dashboard/index.jsx

```diff
@@ -28,7 +28,7 @@ class StudentDashboardShell extends React.Component {
   render() {
     // keep rendering loading screen if the user needs to agree to terms
     // this way the screen stays the same without a flash of other content
-    if (User.terms_signatures_needed) {
+    if (User.unsignedTerms.length) {
```

Comment:
I think this should look at both `terms_signatures_needed` and `unsignedTerms`. It REALLY should match the logic in `isReady` (minus the `isPending`) part.